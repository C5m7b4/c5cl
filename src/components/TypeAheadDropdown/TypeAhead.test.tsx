import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import TypeAheadDropDown from './TypeAheadDropdown';

const FakeData = [
  {
    id: 1,
    storeName: 'Mikes Grocery',
  },
  {
    id: 2,
    storeName: 'Tommy Grocery',
  },
  {
    id: 3,
    storeName: 'TJs Grocery',
  },
  {
    id: 4,
    storeName: 'Marys Grocery',
  },
  {
    id: 5,
    storeName: 'Codys Grocery',
  },
];

describe('TypeAheadDropdown', () => {
  test('should render correctly', () => {
    const fakeFn = jest.fn();
    const { container } = render(
      <TypeAheadDropDown
        data={FakeData}
        placeholder={'Select Store'}
        value={'1'}
        keyValue={'id'}
        displayValue={'storeName'}
        setSelection={fakeFn}
        label={'Stores'}
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('should render with background color and labelColor', () => {
    const fakeFn = jest.fn();
    const { container } = render(
      <TypeAheadDropDown
        data={FakeData}
        placeholder={'Select Store'}
        value={'1'}
        keyValue={'id'}
        displayValue={'storeName'}
        setSelection={fakeFn}
        backgroundColor={'#fff'}
        label={'Stores'}
        labelColor={'#fff'}
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('should search', () => {
    const fakeFn = jest.fn();
    const { container } = render(
      <TypeAheadDropDown
        data={FakeData}
        placeholder={'Select Store'}
        value={'1'}
        keyValue={'id'}
        displayValue={'storeName'}
        setSelection={fakeFn}
        backgroundColor={'#fff'}
        label={'Stores'}
        labelColor={'#fff'}
      />
    );

    const input = container.querySelector('input');
    userEvent.type(screen.getByPlaceholderText(/Select Store/i), 'mi');
    userEvent.type(
      screen.getByPlaceholderText(/Select Store/i),
      '{backspace}{backspace}'
    );
  });
  test('should handle drop down selection', () => {
    const fakeFn = jest.fn();
    const { container, getByRole } = render(
      <TypeAheadDropDown
        data={FakeData}
        placeholder={'Select Store'}
        value={'1'}
        keyValue={'id'}
        displayValue={'storeName'}
        setSelection={fakeFn}
        backgroundColor={'#fff'}
        label={'Stores'}
        labelColor={'#fff'}
      />
    );

    const input = container.querySelector('input');
    userEvent.type(screen.getByPlaceholderText(/Select Store/i), 'mi');

    const option = getByRole('listitem', { name: '' });
    fireEvent.click(option);
  });
  test('should handle no search results', () => {
    const fakeFn = jest.fn();
    const { container, getByRole } = render(
      <TypeAheadDropDown
        data={FakeData}
        placeholder={'Select Store'}
        value={'1'}
        keyValue={'id'}
        displayValue={'storeName'}
        setSelection={fakeFn}
        backgroundColor={'#fff'}
        label={'Stores'}
        labelColor={'#fff'}
      />
    );

    const input = container.querySelector('input');
    userEvent.type(screen.getByPlaceholderText(/Select Store/i), 'di');
  });
  test('should handle selection', () => {
    const fakeFn = jest.fn();
    const { container, getByRole } = render(
      <TypeAheadDropDown
        data={FakeData}
        placeholder={'Select Store'}
        value={'1'}
        keyValue={'id'}
        displayValue={'storeName'}
        setSelection={fakeFn}
        backgroundColor={'#fff'}
        label={'Stores'}
        labelColor={'#fff'}
      />
    );

    const input = container.querySelector('input');
    userEvent.type(screen.getByPlaceholderText(/Select Store/i), 'mi');

    const option = container.querySelector('.tadd-text') as HTMLSpanElement;
    fireEvent.click(option);
  });
  test('should handle onBlur', () => {
    const fakeFn = jest.fn();
    const { container, getByRole } = render(
      <TypeAheadDropDown
        data={FakeData}
        placeholder={'Select Store'}
        value={'1'}
        keyValue={'id'}
        displayValue={'storeName'}
        setSelection={fakeFn}
        backgroundColor={'#fff'}
        label={'Stores'}
        labelColor={'#fff'}
      />
    );

    const input = container.querySelector('input') as HTMLElement;
    userEvent.type(screen.getByPlaceholderText(/Select Store/i), 'mi');

    fireEvent.focusOut(input);
  });
  test('should handle no data', () => {
    const fakeFn = jest.fn();
    const { container, getByRole } = render(
      <TypeAheadDropDown
        data={[]}
        placeholder={'Select Store'}
        value={'1'}
        keyValue={'id'}
        displayValue={'storeName'}
        setSelection={fakeFn}
        backgroundColor={'#fff'}
        label={'Stores'}
        labelColor={'#fff'}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
