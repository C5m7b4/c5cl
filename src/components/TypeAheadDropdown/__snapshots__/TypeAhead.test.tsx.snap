// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeAheadDropdown should handle no data 1`] = `
<div>
  <div
    class="TypeAheadDropDown"
  >
    <div
      style="display: flex; flex-direction: row;"
    >
      <label
        for="typeahead"
        style="color: rgb(255, 255, 255); margin-top: 10px; text-align: right;"
      >
        Stores
      </label>
      <input
        autocomplete="off"
        class="form-control"
        name="typeahead"
        placeholder="Select Store"
        style="background-color: rgb(255, 255, 255);"
        type="text"
        value=""
      />
    </div>
    <ul
      style="position: absolute; left: 7px; top: 0px; width: -10px;"
    />
  </div>
</div>
`;

exports[`TypeAheadDropdown should render correctly 1`] = `
<div>
  <div
    class="TypeAheadDropDown"
  >
    <div
      style="display: flex; flex-direction: row;"
    >
      <label
        for="typeahead"
        style="color: rgb(100, 109, 142); margin-top: 10px; text-align: right;"
      >
        Stores
      </label>
      <input
        autocomplete="off"
        class="form-control"
        name="typeahead"
        placeholder="Select Store"
        style="background-color: rgb(31, 30, 46);"
        type="text"
        value="Mikes Grocery"
      />
    </div>
    <ul
      style="position: absolute; left: 7px; top: 0px; width: -10px;"
    >
      <li>
        <span
          class="tadd-text"
          style="color: white;"
        >
          Mikes Grocery
        </span>
      </li>
      <li>
        <span
          class="tadd-text"
          style="color: white;"
        >
          Tommy Grocery
        </span>
      </li>
      <li>
        <span
          class="tadd-text"
          style="color: white;"
        >
          TJs Grocery
        </span>
      </li>
      <li>
        <span
          class="tadd-text"
          style="color: white;"
        >
          Marys Grocery
        </span>
      </li>
      <li>
        <span
          class="tadd-text"
          style="color: white;"
        >
          Codys Grocery
        </span>
      </li>
    </ul>
  </div>
</div>
`;

exports[`TypeAheadDropdown should render with background color and labelColor 1`] = `
<div>
  <div
    class="TypeAheadDropDown"
  >
    <div
      style="display: flex; flex-direction: row;"
    >
      <label
        for="typeahead"
        style="color: rgb(255, 255, 255); margin-top: 10px; text-align: right;"
      >
        Stores
      </label>
      <input
        autocomplete="off"
        class="form-control"
        name="typeahead"
        placeholder="Select Store"
        style="background-color: rgb(255, 255, 255);"
        type="text"
        value="Mikes Grocery"
      />
    </div>
    <ul
      style="position: absolute; left: 7px; top: 0px; width: -10px;"
    >
      <li>
        <span
          class="tadd-text"
          style="color: white;"
        >
          Mikes Grocery
        </span>
      </li>
      <li>
        <span
          class="tadd-text"
          style="color: white;"
        >
          Tommy Grocery
        </span>
      </li>
      <li>
        <span
          class="tadd-text"
          style="color: white;"
        >
          TJs Grocery
        </span>
      </li>
      <li>
        <span
          class="tadd-text"
          style="color: white;"
        >
          Marys Grocery
        </span>
      </li>
      <li>
        <span
          class="tadd-text"
          style="color: white;"
        >
          Codys Grocery
        </span>
      </li>
    </ul>
  </div>
</div>
`;
